### PR TITLE
Clean tests

### LINKED ISSUE #111: Potential improvements in tests?
Hi @sorhawell, I've been looking a bit at the test suite of `polars` (since I'm not good enough in Rust for now ;)) and I think there are some things there that could be improved. Below is a small list of potential improvements, I prefer posting here first so that (1) we can discuss some of these before spending too much time on this, and (2) other people can contribute to it.

- [x] improve the test coverage. Currently (on my Ubuntu laptop), `covr::report()` says that there is 23.96% of coverage (maybe the "weird" syntax of functions with dollars in name makes coverage detection harder, but my point is that not everything is covered). To me, this is the most important potential improvement but also the one that will take the most time.
- [x] use [snapshots](https://testthat.r-lib.org/articles/snapshotting.html) instead of manually hardcoded expectations. Below is an example of expectation that could be replaced by a snapshot so that it's easier to update if the formatting ever changes.
https://github.com/pola-rs/r-polars/blob/28c3cfeaa2d0834b41e4539e647acec2683a95a2/tests/testthat/test-dataframe.R#L12-L25
- [x] remove uses of `testthat::` (since `testthat` is automatically loaded by `devtools::load_all()`)
- [x] similarly, `polars:::` shouldn't be needed in tests because `devtools::load_all()` also loads internal functions for tests 
- [x] use more suited `expect_` functions, e.g replace `expect_identical(whatever, FALSE)` by `expect_false(whatever)` 
- [x] add some regex to `expect_error()` or `expect_warning()` (arg `regex`) to ensure that the message is really what we expect

There are maybe more but I didn't review all tests for now. As you can see, apart from increasing the coverage, most of these suggestions are here to make the tests easier to read and therefore reduce the cognitive load when something breaks.

Let me know what you think ;)

### PR DESCRIPTION
**Pull request by etiennebacher**
Close #111 

Mostly styling, replacing some expectation functions by other which are more suited, and add some regex for errors whose message is unlikely to change. I gave up styling the chains (cf #113) because there are just too many and there's no automated way to do it (e.g `styler`).

I've been through half of the tests so far and the ~25% coverage reported in #111 seems largely underestimated to me (probably due to the strange construction of functions), there is an amazing amount of tests.

**Still WIP** (stopped at `test-expr_datetime.R`, included), do not pay attention to potential test failures for now

### CODE REVIEW

**Comment by sorhawell** [comment]
@etiennebacher  bare with me, a real big personal hobby-horse coming up :)

I am personally a fan of `data.table` and love that the code base is written predominantly with the `=`assignment operator. I think some parts of the  R community have been pushing the `<-` to an extend, where it is implicitly hinted that the  `=` operator is an inherently defunct operator and a code smell.

I would be proud if some belittled R user one day would try to counter an argument by saying "it is really ok, see r-polars does it too".

How about tests are free game, but the translations predominantly uses `=` ? :)

I searched many time but ever only seen below two code examples to argue "`=`-bad", which I have not yet understood is an issue, because both behaviors are discouraged by the same style guides anyways.

```R
> x=1
> z <- y = x
Error in z <- y = x : could not find function "<-<-"
```
 and because of this
 ```R
 > f = function(x) print(x)
> f(x=2)
[1] 2
> f(x<-4)
[1] 4
> x
[1] 4
```


what do you think? :)

File: tests/testthat/test-bit64.R [hunk 1]
```diff
@@ -1,81 +1,76 @@
...
-    expect_grepl_error(
-      unwrap(polars:::test_robj_to_u32(bit64::as.integer64(-1))),
-      "the value -1 cannot be less than zero"
-    )
+  # NO NA
+  expect_error(
+    unwrap(test_robj_to_i64(bit64::as.integer64(NA))),
+    "scalar arguments do not support integer64 NA value"
+  )
+  expect_error(
```
Comment (sorhawell) [inline tests/testthat/test-bit64.R:50]: If sytematically replacing the `expect_grepl_error` with `expect_error` one need to address that `expect_error` by it self cannot verify r-polars error messages, as the literally expected text will be included in the call also, and the returned error message can never fail. This silently breaks the usefulness of the test.

Options are: rework unwrap() and the provided context of error message or manually turn on/off the same option expect_grepl_error is doing automatically.

**Comment by etiennebacher** [comment]
> How about tests are free game, but the translations predominantly uses = ? :)

I don't have any problems with using `=` instead `<-` in the code (I used to prefer `<-` because that's what I was taught but I saw many popular packages use `=` like `data.table` so I don't really care anymore ;) ).

The thing is simply that I ran `styler::style_dir()` on the tests to fix some styling (add a space after a comma, or before and after a `=`, etc.) and it also replaced `=` by `<-`. I will revert this change at the end, when this PR is ready for review.

File: tests/testthat/test-bit64.R (hunk 1 above)
Comment (etiennebacher) [reply tests/testthat/test-bit64.R:50]: I think I misunderstood you before, I didn't realize that:
```r
# NO NA
expect_error(
    unwrap(test_robj_to_i64(bit64::as.integer64(NA))),
    "hello there"
)
```
would not fail. I'll revert the changes made on `expect_grepl_error()`

**Comment by eitsupi** [comment]
FYI, there is a well-known package `knitr` that uses `=` as the assignment operator, and I believe the `knitr` code is formatted with `formatR`.
https://github.com/yihui/formatR

**Comment by sorhawell** [comment]
I do like formatters. I have not used  it in R yet. I will try it out :)

**Comment by eitsupi** [comment]
I think that the `styler` is probably more popular, although it can probably be configured to use `=` for the assignment operator, but I rarely write settings for `styler` because I have to write the settings in R scripts.
https://stackoverflow.com/questions/50660401/use-instead-of-for-assignment-when-styling-r-code-with-styler

**Comment by vincentarelbundock** [comment]
> I searched many time but ever only seen below two code examples to argue "`=`-bad", which I have not yet understood is an issue, because both behaviors are discouraged by the same style guides anyways.

@sorhawell Here's a video of Ross Ihaka saying we should all be using `=`. Thought you might get a kick out of that 😉 

https://www.youtube.com/watch?v=88TftllIjaY&t=2096s

**Comment by eitsupi** [comment]
I prefer to let the auto-formatter (styler) do the work rather than obsessing over something.
I don't think there is much value in getting hung up on manual formatting here.

**Comment by etiennebacher** [comment]
There's actually not much useful in this PR, most of it is styling and indeed it's better to let `styler` or something else do it. Probably better to close this, there are other priorities (but I let #111 open)

**Comment by sorhawell** [comment]
@vincentarelbundock 

> Thought you might get a kick out of that

Good video also :) I have dream of moving to NZ for two years.

`<-` people and `=` people, all this infighting is unnecessary. We we have so much in common being "left assignment oriented", together we can agree that the "right assignment" `->` people are the odd ones!  😆

**Comment by sorhawell** [comment]
@eitsupi 

I will try to use styler.equals + rstudio addin

https://github.com/Robinlovelace/styler.equals/issues/1

**Comment by eitsupi** [comment]
> @eitsupi
> 
> I will try to use styler.equals + rstudio addin
> 
> [Robinlovelace/styler.equals#1](https://github.com/Robinlovelace/styler.equals/issues/1)

Wonderful! I could not find it.

**Pull request closed without merging**